libs/sdl/sdl.c

```diff
@@ -781,3 +781,21 @@ DEFINE_PRIM(_CURSOR, cursor_create_system, _I32);
 DEFINE_PRIM(_VOID, free_cursor, _CURSOR);
 DEFINE_PRIM(_VOID, set_cursor, _CURSOR);
 DEFINE_PRIM(_ARR, get_devices, _NO_ARG);
+
+// clipboard
+
+HL_PRIM bool HL_NAME(has_clipboard_text)() {
+	return SDL_HasClipboardText();
+}
+
+HL_PRIM const char *HL_NAME(get_clipboard_text)() {
+	return SDL_GetClipboardText();

```

Comment:
Looks like a memory leak to me, the docs say this pointer must be freed with `SDL_free`. We want to copy the bytes into GC-managed memory, then free the original pointer immediately.

Reply:
Not sure how tbh, I'm not used to managed memory.

Reply:
Something like this might work?

```suggestion
	char *text = SDL_GetClipboardText();
	if (!text)
		return NULL;
	vbyte *ret = (vbyte *)hl_to_utf16(text);
	SDL_free(text);
	return ret;
```

Then the returned value should be directly usable for a string, since it is UTF-16 already.